File: labs/python/libs/mti.py
import numpy as np

class Mti(object):
    def __init__(self,s1,s2):
        self._moving_avg_alpha = 0.6
        self._mti_alpha = 0.5
        self._map_avg = np.zeros((s1,s2))
        
    def get_mti_map(self,map):
        mti_map = map - self._map_avg*self._mti_alpha
        # mti_map[mti_map < 0] = np.mean(mti_map)
        self._map_avg = map*self._moving_avg_alpha + self._map_avg*(1-self._moving_avg_alpha)
        return mti_map

File: labs/matlab/libs/Mti.m
classdef Mti < handle
    properties
        moving_avg_alpha;
        mti_alpha;
        map_avg;
    end
    
    methods
        function obj = Mti(s1,s2)
            obj.moving_avg_alpha = 0.8;
            obj.mti_alpha = 1.0;
            obj.map_avg = zeros(s1,s2);
        end
        
        function mti_map = get_mti_map(obj,map)
            mti_map = map - obj.map_avg * obj.mti_alpha;
            mti_map(mti_map <0)= mean(mean(mti_map));
            obj.map_avg = map * obj.moving_avg_alpha + obj.map_avg * (1 - obj.moving_avg_alpha);
        end
    end
end

